src/main/java/org/sagebionetworks/bridge/services/AppConfigService.java

```diff
@@ -112,10 +111,8 @@ public AppConfig getAppConfigForUser(CriteriaContext context, boolean throwExcep
 
         List<AppConfig> appConfigs = getAppConfigs(context.getStudyIdentifier(), false);
 
-        List<AppConfig> matches = appConfigs.stream().filter(oneAppConfig -> {
-            return CriteriaUtils.matchCriteria(context, oneAppConfig.getCriteria());
-        }).sorted(Comparator.comparingLong(AppConfig::getCreatedOn))
-          .collect(BridgeCollectors.toImmutableList());
+        List<AppConfig> matches = CriteriaUtils.filterByCriteria(context, appConfigs);
+        matches.sort(Comparator.comparingLong(AppConfig::getCreatedOn));
```

Comment:
So are we returning the most recent match and *not* sorting by language here? It seems we need to support language sorting here too.